tokio/src/sync/mutex.rs

```diff
@@ -219,6 +219,26 @@ impl<T: ?Sized> Mutex<T> {
         }
     }
 
+    /// Creates a new lock in an unlocked state ready for use.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::sync::Mutex;
+    ///
+    /// static lock: Mutex<i32> = Mutex::const_new(5);
+    /// ```
+    #[cfg(all(feature = "nightly", feature = "parking_lot", not(all(loom, test)),))]
```

Comment:
Why does it require nightly?

Reply:
`Mutex` uses `Semaphore` uses `LinkedList`, `LinkedList` has a trait bound which when using the const-constructor requires `feature(const_fn)` which is nightly see [this comment](https://github.com/tokio-rs/tokio/pull/2790#issuecomment-679164335)

Reply:
I can only see https://github.com/rust-lang/rfcs/pull/2632 as the closest changeset that might let us relieve using `feature(const_fn)`